### PR TITLE
fix: resolve maxTabs auto-close bugs (#6)

### LINKED ISSUE #6: Auto-close on maxTabs threshold not working — multiple bugs identified
## Description

The auto-close when reaching the maxTabs threshold is not working reliably. After tracing the full code execution path from config → storage → background cleanup, I identified several bugs and design issues.

### Bug 1: `handleStorageChanged` does not trigger `applyCleanupRules` after settings save

When user saves settings from the Options page, `chrome.storage.onChanged` fires in the background service worker. The handler `handleStorageChanged()` [only resets the warning icon](https://github.com/pararang/tab-audit/blob/main/src/background/index.ts#L333-L343) — it does **not** call `applyCleanupRules()`.

**Impact**: After changing maxTabs from 50 → 20 and saving, the user must wait up to **1 full minute** for the next alarm tick. Cleanup never runs immediately after config change.

**Fix**: Add `applyCleanupRules()` call in `handleStorageChanged` when `changes.maxTabs` is detected.

### Bug 2: `maxTabs` can become `NaN` via options form (no input validation)

In [options.ts line 132](https://github.com/pararang/tab-audit/blob/main/src/options/options.ts#L132):
```ts
maxTabs: parseInt(elements.maxTabs.value),
```
`parseInt("")` returns `NaN`. If the user clears the field and saves, `NaN` gets stored in `chrome.storage.sync`. Then `getSettings()` returns `maxTabs: NaN` (the spread `{ ...DEFAULT_SETTINGS, ...result }` overwrites the default 50 with NaN).

Since `remainingCount > NaN` is **always `false`** in JavaScript, the max tabs check on [line 233](https://github.com/pararang/tab-audit/blob/main/src/background/index.ts#L233) **never fires**.

**Fix**: Validate input — ensure it is a positive integer; fall back to default if invalid.

### Bug 3: `maxTabs = 0` means "unlimited" per docs, but code closes ALL tabs

[README documents](https://github.com/pararang/tab-audit/blob/main/README.md): "0 = unlimited"
But [line 233](https://github.com/pararang/tab-audit/blob/main/src/background/index.ts#L233):
```ts
if (remainingCount > maxTabs) {
```
With `maxTabs = 0`, this becomes `remainingCount > 0` which is **always true**. The extension will attempt to close ALL non-active, non-whitelisted tabs.

The [test](https://github.com/pararang/tab-audit/blob/main/src/background/index.test.ts#L1252-L1270) only checks "does not crash" — it does **not** verify the "unlimited" behavior (closing nothing).

**Fix**: Add explicit `if (maxTabs <= 0) return;` before the max tabs check block.

### Bug 4: Alarm-based cleanup unreliable in MV3

The extension relies entirely on a 1-minute `chrome.alarms.create("cleanup", { periodInMinutes: 1 })`. In Manifest V3, Chrome can terminate the service worker after ~30s of inactivity. While alarms should wake the worker, there are known Chrome bugs where short-interval alarms (1 min) can be delayed or coalesced, and the service worker may fail to wake properly.

### Bug 5: No pinned tab protection

Pinned tabs (`tab.pinned`) are not excluded from closure. Users who pin tabs expect them to be protected from auto-closing.

## Steps to Reproduce (most likely scenario)

1. Install extension
2. Open Options page → set maxTabs to e.g. 20 → Save
3. Open 30+ tabs across any domains
4. Wait for cleanup… nothing happens for up to 1 minute
5. If the service worker was terminated, alarm may not fire reliably

## Expected Behavior

After saving maxTabs = 20 with 30+ tabs open, cleanup should close the 10+ oldest inactive tabs within a few seconds.

## Environment

- Chrome MV3 (service worker)
- Extension version: 1.4.0
- OS: macOS

## Suggested Priority Fixes

1. Call `applyCleanupRules()` on `handleStorageChanged` for `maxTabs` changes
2. Validate `maxTabs` input in options form (positive integer, fallback to default)
3. Guard `maxTabs <= 0` in the cleanup logic (treat as unlimited)

### PR DESCRIPTION
**Pull request by pararang**
## Summary

Fixes bugs in the maxTabs auto-close feature.

### Changes
- Fixed logic in background service worker for maxTabs auto-close
- Fixed options page handling for maxTabs settings

### Files Changed
- `src/background/index.ts` — corrected auto-close logic
- `src/options/options.ts` — fixed settings handling

## Test Plan
- [x] Extension builds successfully
- [x] Unit tests pass

Closes #6

### CODE REVIEW

File: src/options/options.ts [hunk 1]
```diff
@@ -126,10 +126,18 @@ export async function loadSettingsToForm(elements: OptionsFormElements): Promise
  * @returns Saved settings object
  */
 export async function saveSettingsFromForm(elements: OptionsFormElements): Promise<Settings> {
+  // Validate maxTabs: must be a positive integer, fall back to default if invalid
```
Comment (pararang) [inline src/options/options.ts:129]: ⚠️ **Warning:** Comment says "must be a positive integer" but the code accepts 0 (non-negative integer). The comment should say "non-negative integer" to match the logic on line 131.

File: src/options/options.ts [hunk 2]
```diff
@@ -126,10 +126,18 @@ export async function loadSettingsToForm(elements: OptionsFormElements): Promise
...
+  const parsedMaxTabs = parseInt(elements.maxTabs.value);
```
Comment (pararang) [inline src/options/options.ts:130]: 💡 **Suggestion:** `parseInt("5abc")` returns `5` (partial parse) and `Number.isInteger(5)` is true, so `"5abc"` passes validation. Consider a stricter check like `/^\d+$/` test or `Number(value)` instead. Very minor — unlikely from a number input, but worth noting for robustness.

**Review by pararang — COMMENTED** [review]
## Code Review Summary

**Verdict: Approve** — Solid bug fix PR. Changes are internally consistent, well-tested, no issues.

### Summary
Pinned tab exclusion across all cleanup paths, maxTabs=0 as unlimited, immediate cleanup on settings change, and input validation. +284/-20 across 4 files.

### ⚠️ Warnings
- **options.ts:129** — Comment says "must be a positive integer" but code accepts 0 (non-negative). Misleading — should say "non-negative integer".

### 💡 Suggestions
- **options.ts:130** — `parseInt("5abc")` returns 5 (partial parse) and passes validation. Consider `/^\d+$/` test or `Number(value)` for stricter parsing. Minor — unlikely from a number input field.

### ✅ Looks Good
- Pinned tab exclusion consistent across all 4 cleanup paths
- maxTabs > 0 guard prevents "close everything" edge case
- handleStorageChanged triggers immediate applyCleanupRules
- Options validation catches NaN from empty inputs
- Test coverage covers all new paths and edge cases

---
*Reviewed by Hermes Agent*
